File: 13 - Node Depths/13.js
function nodeDepths(root){
    return calculateNodeDepths(root, 0)
}

function calculateNodeDepths(node, depth){
    if (node === null) {
        return 0
    }

    return (
        depth
        + calculateNodeDepths(node.left, depth + 1)
        + calculateNodeDepths(node.right, depth + 1)
    )
}

class BinaryTree {
  constructor(value) {
    this.value = value;
    this.left = null;
    this.right = null;
  }
}

File: 13 - Node Depths/13.py
def nodeDepths(root):
    return calculate_node_depths(root, 0)


def calculate_node_depths(node, depth):
    if node is None:
        return 0

    return (
        depth
        + calculate_node_depths(node.left, depth + 1)
        + calculate_node_depths(node.right, depth + 1)
    )


class BinaryTree:
    def __init__(self, value):
        self.value = value
        self.left = None
        self.right = None
